src/x/serialize/decoder_fast.go

```diff
@@ -27,40 +27,76 @@ import (
 
 // TagValueFromEncodedTagsFast returns a tag from a set of encoded tags without
 // any pooling required.
+
+type TagVisitor interface {
+	Length(n int)
+	Visit(tag, value []byte) bool
+}
+
 func TagValueFromEncodedTagsFast(
 	encodedTags []byte,
 	tagName []byte,
 ) ([]byte, bool, error) {
+	visitor := findTagValueVisitor{
```

Comment:
Can we be sure this refactoring won't affect the performance of `TagValueFromEncodedTagsFast`?